packages/lambda-api-tracker/src/__test__/xyz-request.test.ts

```diff
@@ -0,0 +1,81 @@
+// process.env['COG_BUCKET'] = 'fake-bucket';
+import { LambdaHttpResponseCloudFront, LambdaSession, LogConfig, LogType } from '@basemaps/lambda-shared';
+import { CloudFrontRequestEvent } from 'aws-lambda';
+import * as o from 'ospec';
+import { LogFn } from 'pino';
+import { handleRequest } from '../index';
+import { ValidateRequest } from '../validate';
+
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+o.spec('xyz-request', () => {
+    const origValidate = ValidateRequest.validate;
+    let mockLogger: LogType;
+
+    /** Generate mock CloudFrontRequestEvent */
+    function req(uri = '', querystring = '?api=12345'): CloudFrontRequestEvent {
+        return {
+            Records: [
+                {
+                    cf: {
+                        request: {
+                            method: 'get',
+                            querystring,
+                            uri,
+                            headers: {
+                                referer: ['from/url'],
+                                'user-agent': ['test browser'],
+                            },
+                            clientIp: '1.2.3.4',
+                        },
+                    },
+                },
+            ],
+        } as any;
+    }
+
+    o.beforeEach(() => {
+        mockLogger = LogConfig.get();
+        mockLogger.info = (o.spy() as unknown) as LogFn;
+    });
+
+    o.after(() => {
+        ValidateRequest.validate = origValidate;
+    });
+
+    o('should catch invalid URLS', async () => {
+        const session = new LambdaSession();
+        const res = await handleRequest(req('/v1/foo'), session, mockLogger);
+
+        o(res.status).equals(404);
+    });
+
+    o('should catch missing api key', async () => {
+        const session = new LambdaSession();
+        const res = await handleRequest(req('/v1/tiles/1/2/3.png', ''), session, mockLogger);
+
+        o(res.status).equals(400);
```

Comment:
do these give more information we can test for eg `res.body`? to validate that it is actually failing due to apikey and not something else?

Reply:
Just the `res.statusDescription`; now added to test for statusDescription. This is just refactored code from index.ts; I can add a body if you like or add a ticket to do so?